cmd/translate/main_client_integration_test.go

```diff
@@ -0,0 +1,314 @@
+//go:build integration
+
+//nolint:paralleltest
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/brianvoe/gofakeit/v6"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.expect.digital/translate/cmd/client/cmd"
+	translatev1 "go.expect.digital/translate/pkg/pb/translate/v1"
+)
+
+func Test_ListServices(t *testing.T) {
+	t.Run("OK", func(t *testing.T) {
+		res, err := cmd.ExecuteWithParams([]string{
+			"service", "ls",
+			"-a", fmt.Sprintf("%s:%s", host, port),
+			"-i", "true",
+		})
+
+		require.NoError(t, err)
+		assert.Contains(t, string(res), "ID")
+	})
+
+	t.Run("error, no transport security set", func(t *testing.T) {
+		res, err := cmd.ExecuteWithParams([]string{
+			"service", "ls",
+			"-a", fmt.Sprintf("%s:%s", host, port),
+		})
+
+		assert.ErrorContains(t, err, "no transport security set")
+		assert.Nil(t, res)
+	})
+}
+
+func Test_TranslationFileUpload(t *testing.T) {
+	t.Run("OK", func(t *testing.T) {
+		service, err := client.CreateService(context.Background(),
+			&translatev1.CreateServiceRequest{Service: randService()})
+
+		require.NoError(t, err)
+		require.NotNil(t, service)
+
+		file, err := os.CreateTemp(t.TempDir(), "test")
+		require.NoError(t, err)
+
+		_, err = file.Write([]byte(`
+		{
+			"language":"lv-lv",
+			"messages":[
+				 {
+					"id":"1",
+					"meaning":"When you greet someone",
+					"message":"hello",
+					"translation":"čau",
+					"fuzzy":false
+				 }
+			]
+	 }`))
```

Comment:
You could use here `randUploadData()` which is definied in `grpc_integration_test.go`, to avoid defining upload data all over again
There are similar functions as well, you could check it out, so tests will become cleaner